### PR TITLE
Update doc for IsRecord

### PR DESCRIPTION
**Pull request by jcouv**
Fixes https://github.com/dotnet/roslyn/issues/69326

### CODE REVIEW

File: src/Compilers/Core/Portable/Symbols/ITypeSymbol.cs [hunk 1]
```diff
@@ -127,7 +127,9 @@ public interface ITypeSymbol : INamespaceOrTypeSymbol
         bool IsReadOnly { get; }
 
         /// <summary>
-        /// True if the type is a record.
+        /// For source symbols, true if the type is a record.
+        /// For metadata symbols, true if the type is a record and a reference type.
+        /// Note: returns false for record structs in metadata since they don't have any distinctive marker.
```
Comment (CyrusNajmabadi) [inline src/Compilers/Core/Portable/Symbols/ITypeSymbol.cs:132]: could be in a remarks section.

**Review by CyrusNajmabadi — APPROVED** [review]

**Comment by jcouv** [comment]
@dotnet/roslyn-compiler for a small xml doc change. Thanks

File: src/Compilers/Core/Portable/Symbols/ITypeSymbol.cs (hunk 1 above)
Comment (333fred) [reply src/Compilers/Core/Portable/Symbols/ITypeSymbol.cs:132]: Yes, this should be in a `remarks` section.

**Review by 333fred — APPROVED** [review]
One small comment, otherwise LGTM

**Pull request merged**